### ISSUE TITLE
Add Pros and Cons for Cache Location (session storage vs. local storage)

### ISSUE THREAD

**Issue by svdHero** [AzureAD/microsoft-authentication-library-for-js#5221]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Wrapper Library

MSAL React (@azure/msal-react)

### Public or Confidential Client?

Public

### Documentation Location

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/caching.md#cache-storage

### Description

Why has Microsoft chosen session storage over local storage as a default value for the MSAL cache location? What are the disadvantages if I change the cache location to local storage?

All I could find online was just [this overview table](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/caching.md#cache-storage). Would it be possible to add the pros and cons (i.e. risks) for the various cache locations to said document?

For context: The question arose when adding end-to-end tests using Playwright to my React SPA. I ran into problems when trying to save and restore browser storage snapshots, see [Playwright documentation](https://playwright.dev/docs/test-auth#multiple-signed-in-roles). In a nutshell, Playwright can only capture local storage state but not session storage state, which causes my end-to-end tests to fail, because MSAL loses the authentication cache. For more details see https://github.com/microsoft/playwright/issues/17328.

**Comment by sameerag** [comment]
@svdHero The difference between these storage types is exactly as the MDN docs mention it [here](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API). 

We chose to make `sessionStorage` default as it has  less persistence among the two. However choosing a specific storage totally depends on your application model and how you want the cache to persist across tabs, across sessions and in between browser close/reopen .

We will take your feedback and add this context more clearly in our samples. 
cc @derisen do you want to address this?

**Comment by derisen** [comment]
@svdHero cache location choice is essentially a tradeoff between user experience vs. security. Since session storage has shorter lifespan and not shared between tabs, it creates smaller attack surface. In general, using the session storage whenever possible (instead of local storage) is a security [best practice](https://cheatsheetseries.owasp.org/cheatsheets/HTML5_Security_Cheat_Sheet.html). So the pros of session storage is smaller attack surface, while cons are no shared auth state between tabs and the cache is gone when the user closes the browser (so in the end increased likelihood of bothering the user with more prompts). Let me see if we can emphasize this more clearly in documentation.

Just to note, MSAL will always use session storage to cache some temporary auth artifacts (like current interaction status and etc.), unless you opt-out of using browser storage completely by setting your `cacheLocation` to `memoryStorage`. So even if you set to use `localStorage`, there will be some items in the `sessionStorage`!

Please let us know of your experience with Playwright. It could be something we would like to offer a recipe to other devs.

**Comment by sameerag** [comment]
@derisen Do we need to document this?

**Comment by svdHero** [comment]
@derisen thank you for your help and the explanation. I would rather like to stay with session storage and I am not happy with Playwright with regard to this feature. I will open a feature request with them for an option to include session storage in storage export. I will link the issue here.

Overall, I am quite happy with Playwright in combination with MSAL. At the moment, I just repeat the sign-in process before each test, which works fine.
I only noticed that I had to carefully tweak a few element selectors for testing the B2C login forms in Safari. Apparently B2C seems to generate different html code for Safari than for Chrome or Firefox. That was a PITA. But that's unrelated to MSAL.
